Riot/Modules/Common/SwiftUI/VectorHostingBottomSheetPreferences.swift

```diff
@@ -25,11 +25,23 @@ class VectorHostingBottomSheetPreferences {
         case medium
         case large
         
+        /// only available on iOS16, medium behaviour will be used instead
+        case custom(height: CGFloat, identifier: String = "custom")
```

Comment:
```suggestion
        case custom(height: () -> CGFloat?, identifier: String?)
```
It could be handy leveraging optional height to disable detents when using this API.
Also the identifier should be optional to match Apple's api.
Probably you also may need to make optional the return value of `func uiSheetDetentId()` but it shouldn't be a problem because it's used where optionals are expected.

Reply:
I understand your reasoning but in my opinion doing so you still remove flexibility in our api.
Changing the API like I suggest you have dynamic detents behaviors like this:
```swift
let detents:  [VectorHostingBottomSheetPreferences.Detent] = [
     .medium,
     . custom(
         height: { 
              guard someCondition else {
                  return nil
              }
              return someComputationReturningCGFloat()
          }, identifier: nil)
]
```
And I think the effort should be quite small.

Reply:
dynamic detents would require to also use the function `invalidateDetents`, because custom detents are only recalculated if such function is called as explained in the docs here:
https://developer.apple.com/documentation/uikit/uisheetpresentationcontroller/detent/3976719-custom

In fact if the calculation depends on a specific value that is stored and then this value changes, nothing would happen until the detents are invalidated.

It's more effort than what it seems to have dynamic custom detents, and I don't think is worth the effort for this level of flexibility, since it would impact the overall readibility of the wrapper.

Reply:
I was thinking mainly when you start presenting a new vc.
In that case you don't need to invalidate since the presentation didn't start yet.
Btw you could do the same with your proposed implementation returning different array of detents.
So IGTM ;-)